src/test/java/com/jcraft/jsch/JSchTest.java

```diff
@@ -64,6 +97,44 @@ void checkLoggerBehavior() throws Exception {
         assertSame(JSch.DEVNULL, jsch.getInstanceLogger(), "instance logger should be DEVNULL");
     }
     
+    @Test
+    void checkFillConfig() throws Exception {
+      // TODO add more tests, this rudimentary implementation shows the
+      // reason for the javaVersion-parameter discussed in #130
+      
+      Properties orgProps = System.getProperties();
+      Properties props = new Properties();
+      try {
+        System.setProperties(props);
+        
+        HashMap<String, String> map = new HashMap<>();
+        
+        JSch.fillConfig(map, 10);
+        assertEquals("com.jcraft.jsch.bc.XDH", map.get("xdh"), "check of xdh failed");
+        JSch.fillConfig(map, 11);
+        assertEquals("com.jcraft.jsch.jce.XDH", map.get("xdh"), "check of xdh failed");
+        JSch.fillConfig(map, 12);
```

Comment:
Do we need to test this for `12` since it is the same as `11`?

Reply:
I'm always testing the edge cases to make sure that if somebody for some reason changes `>=` to `==`, the test fails.

Reply:
To `-v` that: `a >= b` can be seen as a shortcut for `a == b || a > b`. To have complete coverage over the conditions you need to implement three tests: one for `a < b`, one for `a == b` and one for `a > b`.